### DISCUSSION TITLE
Phantom API to import a wallet

### DISCUSSION THREAD

**Discussion by cbuckley-code** [phantom/docs#257] [General]
So I was thinking about how to onboard normies easier.  I see a number of projects are setting up custodian wallets (tiplink.io and deadcaster.xyz) for example and hide the wallet for the end user; I know it's hard for a lot of people to grasp wallets and swapping one token for another and paying fees in one token to get another token.  The custodian wallets help but, also don't allow full access to a lot of things.  

If Phantom would allow me to import a mnemonic into the extension or mobile app I think I could really simplify a lot of things for my users.  Right now when someone signs up on deadcaster.xyz I have a background function that creates an associated token account for my token (I'm using Solana...) and I encrypt and store their mnemonic and generate a Keypair from it.  I give them a page with the private key and mnemonic  that they can copy and then open the Phantom wallet and paste it in but, I'd like to be able to help that process out and ask them to install the Phantom extension (or app) and if it's available have a ux that guides them export their deadcaster wallet to Phantom (which allows them to swap on jupiter, allow blinks, etc.) and our code would pass the mnemonic to Phantom then remove the mnemonic from our servers.

Thoughts?  I know there is the chrome-extension://---/onboarding.html?append=true page maybe I can just leverage that?

Thank you,

Chris

**Comment by adamdelphantom** [comment]
Hey @cbuckley-code,

Thank you for writing in! I don't have an eta to provide but I believe much of the ux you describe will be enabled by the offerings that will come out of this [recent announcement.](https://x.com/phantom/status/1792903231774834689). There's also a lengthy discussion about the topic [here.](https://blockworks.co/podcast/lightspeed/2fcdbb04-192a-11ef-8bbb-b334c296481a)

**Reply by cbuckley-code** [reply]
Thank you.
